internal/armada/server/submit.go

```diff
@@ -454,6 +454,10 @@ func (server *SubmitServer) checkQueuePermission(
 	return nil, groups
 }
 
+// brought outside for mocking.
+var NewULID = util.NewULID
```

Comment:
I don't love this style. Feels like hack and messes up the code for the sake of testing.

Considering createJobs could be extracted out of submit entirely. Possibly it could be some struct should handle createJobs, it can then be configured with a uuidFunc and timeFunc. 

At the very least, it shouldn't be exported

Reply:
fixed the exporting. Going to leave this like this for today at least and maybe come back to it tomorrow